clients/cobol-lsp-vscode-extension/src/services/JavaCheck.ts

```diff
@@ -21,12 +21,13 @@ export class JavaCheck {
     }
     public async isJavaInstalled() {
         return new Promise<any>((resolve, reject) => {
+            let resolved = false;
             const ls = cp.spawn("java", ["-version"]);
             ls.stderr.on("data", (data: any) => {
-                if (!JavaCheck.isJavaVersionSupported(data.toString())) {
-                    reject("Minimum expected Java version is 8");
+                if (JavaCheck.isJavaVersionSupported(data.toString())) {
```

Comment:
What happens if not supported ? Like an else case.

Reply:
there will be a reject
https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/396/files#diff-f4a7f7595d05dd5c4124b39716587168R42

Reply:
But it is not on "data"

Reply:
It can't be in data. Data is called for each line. And we don't know which line contains a version number.
If there was no good version in any of line, then JavaCheck rejects promise on close.

Reply:
In this case resolve() should be out of if block, otherwise it is confusing

Reply:
Sorry, I didn't get it.. which block?

Reply:
I mean, negative test-scenario is missing here

Reply:
Added one